### PR TITLE
Lexer progress 

### PR DESCRIPTION
**Pull request by Carcass-Project**
hello <3

### CODE REVIEW

**Comment by mauro-balades** [comment]
Tell me when you finished and remember to follow the ts code style with camelCase. 👍

**Comment by Carcass-Project** [comment]
imagine camelCase

**Comment by mauro-balades** [comment]
@Carcass-Project can you please push the lexer so we can test things out?

**Comment by hrszpuk** [comment]
> Maybe he needs help implementing it? I know he's learning TypeScript at the same time as developing it, so maybe someone can jump in with an assist?

Biggest issue with this is it's a fork, if it was a branch we would have permissions to push commits to it...

**Comment by hrszpuk** [comment]
I just think it's better to use feature branching apposed to forking :P

**Comment by mauro-balades** [comment]
let ma boy checl

**Comment by mauro-balades** [comment]
There are some git conflicts that I would need to solve when I get home.

File: src/lexer/index.ts [hunk 1]
```diff
@@ -1,22 +1,152 @@
...
+      this.next();
     }
 
-    public get tokens() {
-        return this._tokens;
+    switch (fullIdent) {
+      case 'Func':
+        return new Token(TT.KW_FUNC, fullIdent);
+      case 'Event':
+        return new Token(TT.KW_EVENT, fullIdent);
```
[the remark this answers is not in the document]
Comment (Carcass-Project) [reply src/lexer/index.ts:102]: idk, nobody told me about those specifics

[the remark this answers is not in the document]
Comment (hrszpuk) [reply src/lexer/index.ts:102]: > idk, nobody told me about those specifics

I'm going to end up making a pseudo-specification 😔

**Review by hrszpuk — COMMENTED** [review]
Amazing merge mauro

**Pull request merged**